x/axelarnet/types/msg_call_contract.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+)
+
+// NewCallContractRequest is the constructor for NewCallContractRequest
+func NewCallContractRequest(sender sdk.AccAddress, chain string, contractAddress string, payload []byte) *CallContractRequest {
+	return &CallContractRequest{
+		Sender:          sender,
+		Chain:           nexus.ChainName(utils.NormalizeString(chain)),
+		ContractAddress: contractAddress,
+		Payload:         payload,
+	}
+}
+
+// Route returns the route for this message
+func (m CallContractRequest) Route() string {
+	return RouterKey
+}
+
+// Type returns the type of the message
+func (m CallContractRequest) Type() string {
+	return "CallContract"
+}
+
+// ValidateBasic executes a stateless message validation
+func (m CallContractRequest) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(m.Sender); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, sdkerrors.Wrap(err, "sender").Error())
+	}
+
+	if err := m.Chain.Validate(); err != nil {
+		return fmt.Errorf("invalid chain %s: %s", m.Chain, err)
+	}
+
```

Comment:
Maybe we want to allow more than 128 chars for contract address size, in case in the future destination chain is some non-evm chain with longer addresses?

Reply:
agreed, this should be a check on the destination side, not a preemptive safeguard here

Reply:
I just validated that the address is not empty

Reply:
we can extend that when using wasm if needed. but in the consensus, we don't want to store arbitrarily long values

Reply:
we could impose some arbitrary limit that's quite large just to prevent spam, but is unlikely to ever be too small for an actual address. Say 512 chars or 1024 or some such